src/scenic/core/object_types.py

```diff
@@ -3,666 +3,1760 @@
 Defines the 3 Scenic classes `Point`, `OrientedPoint`, and `Object`, and associated
 helper code (notably their base class `Constructible`, which implements the handling of
 property definitions and :ref:`specifier resolution`).
+
+.. warning::
+
+    In 2D compatibility mode, these classes are overwritten with 2D analogs. While
```

Comment:
The wording here is good but let's make "2D compatibility mode" a cross-reference to the docs page describing it.